### PR TITLE
[Polygon] Updated paraswap to V5

### PR DESCRIPTION
**Pull request by bhavik-m**
- [x] deployed and comment address [polygon]
- [x] Add to dsa-connect [here](https://github.com/Instadapp/dsa-connect/pull/117)

### CODE REVIEW

**Review by thrilok209 — APPROVED** [review]

File: test/polygon/paraswap/paraswap.test.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  const [wallet0, wallet1] = wallets;
+  before(async () => {
+    await hre.network.provider.request({
+        method: "hardhat_reset",
+        params: [
+            {
+                forking: {
+                    // @ts-ignore
+                    jsonRpcUrl: hre.config.networks.hardhat.forking.url,
+
```
Comment (pradyuman-verma) [inline test/polygon/paraswap/paraswap.test.ts]: Remove this unnecessary blank line!

File: test/polygon/paraswap/paraswap.test.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+import BigNumber from "bignumber.js";
+import axios from "axios";
+const { waffle, ethers } = hre;
+const { provider, deployContract } = waffle;
+describe("Paraswap", function() {
+  const connectorName = "paraswap-test";
+  let dsaWallet0: any;
+  let masterSigner: any;
+  let instaConnectorsV2: any;
+  let connector: any;
```
Comment (pradyuman-verma) [inline test/polygon/paraswap/paraswap.test.ts]:
```suggestion
  let dsaWallet0: Signer;
  let masterSigner: Signer;
  let instaConnectorsV2: Contract;
  let connector: Contract;
```

File: test/polygon/paraswap/paraswap.test.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+          srcDecimals: sellTokenDecimals,
+          destDecimals: buyTokenDecimals,
+          srcAmount: srcAmount,
+          destAmount: minAmount,
+          userAddress: fromAddress,
+        };
+        let url2 = "https://apiv5.paraswap.io/transactions/137?ignoreChecks=true";
+        const calldata = await axios
+          .post(url2, txConfig)
+          .then((data) => data.data.data);
```
Comment (pradyuman-verma) [inline test/polygon/paraswap/paraswap.test.ts]:
```suggestion
        const calldata = (await axios
          .post(url2, txConfig)).data.data
```

File: test/polygon/paraswap/paraswap.test.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+          srcDecimals: sellTokenDecimals,
+          destDecimals: buyTokenDecimals,
+          amount: srcAmount,
+          side: "SELL",
+          network: 137,
+        };
+
+        const priceRoute = await axios
+          .get(url, { params: params })
+          .then((data) => data.data.priceRoute);
```
Comment (pradyuman-verma) [inline test/polygon/paraswap/paraswap.test.ts]:
```suggestion
        const priceRoute = (await axios
          .get(url, { params: params })).data.priceRoute
```

File: test/polygon/paraswap/paraswap.test.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        let url = `https://apiv5.paraswap.io/prices/`;
+        let params = {
+          srcToken: sellTokenAddress,
+          destToken: buyTokenAddress,
+          srcDecimals: sellTokenDecimals,
+          destDecimals: buyTokenDecimals,
+          amount: srcAmount,
+          side: "SELL",
+          network: 137,
+        };
```
Comment (pradyuman-verma) [inline test/polygon/paraswap/paraswap.test.ts:94]: I think they have changed the params for v5. 
check it here - https://developers.paraswap.network/api/v4-to-v5#1.-prices-endpoint

**Comment by pradyuman-verma** [comment]
## Deployed on Polygon:

**Name:** `PARASWAP-V5-A`
**Address:** [0x7f4466Faf06A2CA3b10e806cC21E2280E5Ef3D8f](https://polygonscan.com/address/0x7f4466faf06a2ca3b10e806cc21e2280e5ef3d8f#code)
**tx:** [here](https://polygonscan.com/tx/0xf8f91c888c92a1aa5d055ab1be4067fc29d20f728309a1c8cb51be1cb9fc1e4b)

**Pull request merged**